tests/utilities/test_plugins.py

```diff
@@ -0,0 +1,60 @@
+import inspect
+import shutil
+import unittest
+from pathlib import Path
+from unittest.mock import patch
+import os
+import importlib.util
+
+import jinja2
+
+from dagger.utilities.module import Module  # Adjust this import according to your actual module structure
+from dagger import conf
+
+TESTS_ROOT = Path(__file__).parent.parent
+
+class TestLoadPlugins(unittest.TestCase):
+
+    def setUp(self):
+        self._jinja_environment = jinja2.Environment()
+        loaded_classes = Module.load_plugins()
+        for class_name, class_obj in loaded_classes.items():
+            self._jinja_environment.globals[class_name] = class_obj
```

Comment:
This part is copying code from the original implementation. Can't we just structure the code better and call the actual implementation?

Reply:
changed